core-web/libs/edit-content/src/lib/fields/dot-edit-content-wysiwyg-field/dot-wysiwyg-plugin/dot-wysiwyg-plugin.service.spec.ts

```diff
@@ -0,0 +1,195 @@
+import { expect } from '@jest/globals';
+import { createServiceFactory, SpectatorService } from '@ngneat/spectator';
+import { MockComponent } from 'ng-mocks';
+import { of } from 'rxjs';
+
+import { DialogService, DynamicDialogRef } from 'primeng/dynamicdialog';
+
+import { DotUploadFileService } from '@dotcms/data-access';
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+import { DotAssetSearchDialogComponent } from '@dotcms/ui';
+import { EMPTY_CONTENTLET } from '@dotcms/utils-testing';
+
+import { DotWysiwygPluginService } from './dot-wysiwyg-plugin.service';
+import { formatDotImageNode } from './utils/editor.utils';
+
+/**
+ * This Mock is used to check we are sending the correct configuration to the editor
+ * No need to mock all the methods and properties of the Editor
+ * Some methods are customized to check the configuration
+ */
+class MockEditor {
+    private customButtons = {};
+    private events = {};
+
+    ui = {
+        registry: {
+            getAll: () => {
+                return {
+                    buttons: this.customButtons
+                };
+            },
+            addButton: (name, config) => {
+                this.customButtons[name] = config;
+            }
+        }
+    };
+
+    on = (name, fn) => {
+        if (!this.events[name]) {
+            this.events[name] = [fn];
+
+            return;
+        }
+
+        this.events[name].push(fn);
+    };
+
+    fakeOnCall = (name, event) => {
+        this.events[name].forEach((fn) => fn(event));
+    };
+
+    insertContent = jest.fn();
+}
+
+describe('DotWysiwygPluginService', () => {
+    let spectator: SpectatorService<DotWysiwygPluginService>;
+    let dialogService: DialogService;
+    let dotUploadFileService: DotUploadFileService;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let editor: any;
```

Comment:
let editor: InstanceType<typeof MockEditor>
Maybe this to avoid the any

Reply:
@KevinDavilaDotCMS  `any` is used here because the Editor is a complex object that we don't need to mock all the methods and properties. We are using this mock to check the configuration of the editor. This mock also contains some custom methods to check the configuration.

Reply:
I'll add a comment